internal/config/registry.go

```diff
@@ -0,0 +1,68 @@
+package config
+
+import (
+	"os"
+
+	"github.com/anchore/stereoscope/pkg/image"
+
+	"github.com/spf13/viper"
+)
+
+type RegistryCredentials struct {
+	Authority string `yaml:"authority" json:"authority" mapstructure:"authority"`
+	// IMPORTANT: do not show the username in any YAML/JSON output (sensitive information)
+	Username string `yaml:"-" json:"-" mapstructure:"username"`
+	// IMPORTANT: do not show the password in any YAML/JSON output (sensitive information)
+	Password string `yaml:"-" json:"-" mapstructure:"password"`
+	// IMPORTANT: do not show the token in any YAML/JSON output (sensitive information)
+	Token string `yaml:"-" json:"-" mapstructure:"token"`
+}
+
+type registry struct {
+	InsecureSkipTLSVerify bool                  `yaml:"insecure-skip-tls-verify" json:"insecure-skip-tls-verify" mapstructure:"insecure-skip-tls-verify"`
+	Auth                  []RegistryCredentials `yaml:"auth" json:"auth" mapstructure:"auth"`
+}
+
+func (cfg registry) loadDefaultValues(v *viper.Viper) {
+	v.SetDefault("registry.insecure-skip-tls-verify", false)
+	v.SetDefault("registry.auth", []RegistryCredentials{})
+}
+
+func (cfg *registry) parseConfigValues() error {
+	// there may be additional credentials provided by env var that should be appended to the set of credentials
+	// note: only a authority and password would be required since a token
+	authority, username, password, token :=
+		os.Getenv("SYFT_REGISTRY_AUTH_AUTHORITY"),
+		os.Getenv("SYFT_REGISTRY_AUTH_USERNAME"),
+		os.Getenv("SYFT_REGISTRY_AUTH_PASSWORD"),
+		os.Getenv("SYFT_REGISTRY_AUTH_TOKEN")
+
+	if authority != "" && password != "" {
```

Comment:
if I'm configuring this with a token only then this condition wouldn't happen, is that an oversight? or when would a token be prepended?

Reply:
excellent call, adjusting now. I'm going to see if I can get a cli-level test to cover this

Reply:
cli test added for asserting registry options in general